api/types/installers/installer.sh.tmpl

```diff
@@ -3,10 +3,9 @@
 set -eu
 
 on_ec2() {
-  EC2_STATUS=$(curl -o /dev/null -w "%{http_code}" -m5 -sS "http://169.254.169.254/latest/meta-data")
-  # EC2 metadata sometimes requires token access, so a successful hit may
-  # return unauthorized/forbidden.
-  [ "$EC2_STATUS" = "200" ] || [ "$EC2_STATUS" = "401" ] || [ "$EC2_STATUS" = "403" ]
+  IMDS_TOKEN=$(curl -m5 -sS -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 300")
+  EC2_STATUS=$(curl -o /dev/null -w "%{http_code}" -m5 -sS -H "X-aws-ec2-metadata-token: ${IMDS_TOKEN}" "http://169.254.169.254/latest/meta-data")
```

Comment:
If this isn't on EC2, `IMDS_TOKEN` will be filled with junk. It might be worth checking the status of that request too.

Reply:
Good catch. Fixed!